scripts/add_electricity.py

```diff
@@ -286,6 +273,67 @@ def update_transmission_costs(n, costs, length_factor=1.0, simple_hvdc_costs=Fal
             + costs.at["HVDC inverter pair", "capital_cost"]
         )
     n.links.loc[dc_b, "capital_cost"] = costs
+    return n
+
+
+def attach_dc_costs_lines(n, costs, length_factor=1.0, simple_hvdc_costs=False):
+    dc_b = n.lines.carrier == "DC"
+    if simple_hvdc_costs:
+        costs = (
+            n.lines.loc[dc_b, "length"]
+            * length_factor
+            * costs.at["HVDC overhead", "capital_cost"]
+        )
+    else:
+        costs = (
+            n.lines.loc[dc_b, "length"]
+            * length_factor
+            * (
+                (1.0 - n.lines.loc[dc_b, "underwater_fraction"])
+                * costs.at["HVDC overhead", "capital_cost"]
+                + n.lines.loc[dc_b, "underwater_fraction"]
+                * costs.at["HVDC submarine", "capital_cost"]
+            )
+            + costs.at["HVDC inverter pair", "capital_cost"]
+        )
+    n.lines.loc[dc_b, "capital_cost"] = costs
+    return n
+
+
+def update_transmission_costs(n, costs, length_factor=1.0, simple_hvdc_costs=False):
+    n.lines["capital_cost"] = (
+        n.lines["length"] * length_factor * costs.at["HVAC overhead", "capital_cost"]
+    )
+
+    if n.links.empty:
+        return
+
+    # If there are no "DC" links, then the 'underwater_fraction' column
+    # may be missing. Therefore we have to return here.
+    # TODO: Require fix
+    if (
+        n.links.loc[n.links.carrier == "DC"].empty
+        and n.lines.loc[n.lines.carrier == "DC"].empty
+    ):
+        return
+
+    # HVDC implemented as links
+    if not n.links.loc[n.links.carrier == "DC"].empty:
+        attach_dc_costs_links(
```

Comment:
To add clarity, if the general function attach_dc_costs is feasible and theempty check could be moved inside, we could do two lines, like:
`attach_dc_costs(n.lines, ...)`
`attach_dc_costs(n.links, ...)`
but that would be a dream... not sure if it is feasible.
I think your experience here is highly

Reply:
Done! :)